carddav/server.go

```diff
@@ -242,20 +244,28 @@ type backend struct {
 	Backend Backend
 }
 
-func (b *backend) Options(r *http.Request) (caps []string, allow []string, err error) {
-	caps = []string{"addressbook"}
+type resourceType int
 
-	homeSetPath, err := b.Backend.AddressbookHomeSetPath(r.Context())
-	if err != nil {
-		return nil, nil, err
-	}
+const (
+	resourceTypeRoot               resourceType = iota
+	resourceTypeUserPrincipal      resourceType = iota
+	resourceTypeAddressBookHomeSet resourceType = iota
+	resourceTypeAddressBook        resourceType = iota
+	resourceTypeAddressObject      resourceType = iota
```

Comment:
Nit: only the first `resourceType = iota` is necessary, the rest can be omitted.